src/main/java/org/opensearch/neuralsearch/transport/MLPredictTransportAction.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.transport;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.neuralsearch.ml.MLCommonsClientAccessor;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+/**
+ * Transport action to do call inference/predict api of ML Client.
+ */
+public class MLPredictTransportAction extends HandledTransportAction<MLPredictActionRequest, MLPredictActionResponse> {
```

Comment:
MLCommons Transport action is an internal action of MLCommons. The MLClient provides abstraction to use those actions. I added this Transport action so that if Neural Plugin wants call the MLClient APIs they have a way to do so. Otherwise, they can directly call the MLCommonsAccessor. These are 2 ways to call the same API.

Reply:
I dont see a case where in this plugin, we wouldnt always want to use the MLCommonsAccessor. Can you think of any cases?

Reply:
You can directly use from the query clause if you want to, or ingestion APIs can use. Its upto us how we want to use it.

Reply:
Let's keep this for this PR, if it won't get used I will remove it.

Reply:
I was thinking opposite - remove it from this PR and if there is a use case that requires it, we can add it. This way, we simplify the code.

The only case where I could think we potentially want to use a transport action is if we need to forward the request. However, in the confines of the search/processor extension points, we cannot do this.

Reply:
I will still like to keep it for  now. There is no harm. I will remove this once we agree that we don't need this in a later PR.

Reply:
An

Reply:
Ok